blocks/settlement.go

```diff
@@ -205,12 +207,16 @@ func LastToSettleAt(settleAt uint64, parent *Block) (b *Block, ok bool) {
 	// settled (see invariant in [Block]). If a block was already settled then
 	// only it or a later (i.e. unsettled) block can be returned by this loop,
 	// therefore we have a guarantee that the loop update will never result in
-	// `block==nil`.
+	// `block==nil` during block building. Verification, however, MUST protect
+	// against malformed input so performs a check.
 	for block := parent; ; block = block.ParentBlock() {
+		if block == nil {
+			return nil, false, fmt.Errorf("%w: settling at %d with parent %#x (%v)", errIncompleteBlockHistory, settleAt, parent.Hash(), parent.Number())
+		}
```

Comment:
Removing this check does not cause any tests to fail. If this is actually needed (which I don't think it is), we should definitely add a regression test for it.